oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -253,16 +260,36 @@ public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bu
                     LOG.error("Error decoding bulk response", e);
                 }
             }
+            NodeBuilder status = definitionBuilder.child(IndexDefinition.STATUS_NODE);
             if (bulkResponse.hasFailures()) { // check if some operations failed to execute
+
+                // Read the current failed paths (if any) on the :status node into failedDocList
+                if (status.getProperty(IndexDefinition.FAILED_DOC_PATHS) != null) {
+                    for (String str : status.getProperty(IndexDefinition.FAILED_DOC_PATHS).getValue(Type.STRINGS)) {
+                        failedDocSet.add(str);
+                    }
+                }
+
                 for (BulkItemResponse bulkItemResponse : bulkResponse) {
                     if (bulkItemResponse.isFailed()) {
                         BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
-                        LOG.error("Bulk item with id {} failed", failure.getId(), failure.getCause());
+                        failedDocSet.add(bulkItemResponse.getId());
+                        // Log entry to be used to parse logs to get the failed doc id/path if needed
+                        LOG.error("ElasticIndex Update Doc Failure: Error while adding/updating doc with id : [{}]", bulkItemResponse.getId());
+                        LOG.error("Failure Details: BulkItem ID: {}, Failure Cause: {}", failure.getId(), failure.getCause());
                     } else {
                         // Set indexUpdated to true even if 1 item was updated successfully
                         updatesMap.put(executionId, Boolean.TRUE);
                     }
                 }
+
+                if (failedDocSet.size() > FAILED_DOC_COUNT_FOR_STATUS_NODE) {
```

Comment:
Keeping large number of strings in a single property doesn't scale well. I don't know what is the threshold but I have seen problems before on tarmk. Few thousand entries resulted in bloating of tars on every update.
Is converting this property to binary (after a threshold is reached) an option? I don't know if breaking the property into multiple properties or across multiple nodes would help here.
Using these approaches we might be able to add all failing documents here.

Reply:
I am not sure if we can actually change the property type on runtime (I don't think we can). What about we set this to Type.BINARY from beginning only and store failedDocSet.toString as it's value ? wdyt ?

Reply:
@averma21  - I looked at this again. This property would only get updated in case there are any failures in the bulk response, and not every time an index update runs. So there might not be a case where we have a lot of updates to this property.

Even if there is such a case, that would suggest something seriously wrong with elastic server (it's probably down or maybe getting disconnected very frequently) - I suppose that would already raise an alarm and segment bloating might not be our biggest problem then (since that will anyhow get back to normal size after online compaction - right ? )

Reply:
@nit0906  
> might not be a case where we have a lot of updates to this property
yes that should be true in this case. However there could be multiple updates in case there is a problem during certain cases like bulk ingestion of assets, but given that such problems should be rare, I think we should be fine.